src/v2/Apps/Onboarding/useOnboardingContext.tsx

```diff
@@ -2,48 +2,87 @@ import {
   FC,
   createContext,
   createRef,
-  useCallback,
   useRef,
   useContext,
-  useState,
-  Dispatch,
-  SetStateAction,
+  useReducer,
+  useEffect,
 } from "react"
 import { WorkflowEngine } from "v2/Utils/WorkflowEngine"
 import { useConfig } from "./config"
 
-type Answer = string | null
+type State = {
+  questionOne: string | null
+  questionTwo: string[]
+  questionThree: string | null
+}
 
-export type Basis = {
-  answer: Answer
+export const DEFAULT_STATE: State = {
+  questionOne: null,
+  questionTwo: [],
+  questionThree: null,
 }
 
-export const DEFAULT_BASIS: Basis = {
-  answer: null,
+type Action =
+  | { type: "RESET" }
+  | { type: "SET_ANSWER_ONE"; payload: string }
+  | { type: "SET_ANSWER_TWO"; payload: string }
+  | { type: "SET_ANSWER_THREE"; payload: string }
+
+const reducer = (onReset: () => void) => (state: State, action: Action) => {
+  switch (action.type) {
+    case "RESET":
+      onReset()
+      return DEFAULT_STATE
+
+    case "SET_ANSWER_ONE":
+      return {
+        ...state,
+        questionOne: action.payload,
+      }
+
+    case "SET_ANSWER_TWO":
+      return {
+        ...state,
+        questionTwo: state.questionTwo.includes(action.payload)
+          ? state.questionTwo.filter(answer => answer !== action.payload)
+          : [...state.questionTwo, action.payload],
+      }
+
+    case "SET_ANSWER_THREE":
+      return {
+        ...state,
+        questionThree: action.payload,
+      }
+
+    default:
+      return state
+  }
 }
 
+/**
+ * Basis is just State stored in a ref.
+ * It is used to make decisions in the workflow.
+ */
+export type Basis = State
```

Comment:
Basis just becomes a simple copy of State; just stuck in a ref so that the workflow engine can access it.